lib/msal-common/src/request/RequestParameterBuilder.ts

```diff
@@ -608,6 +612,35 @@ export class RequestParameterBuilder {
             queryParameterArray.push(`${key}=${value}`);
         });
 
+        const correlationId = this.parameters.get(
+            AADServerParamKeys.CLIENT_REQUEST_ID
+        );
+        const clientId = this.parameters.get(AADServerParamKeys.CLIENT_ID);
+
+        if (
+            this.parameters.has(AADServerParamKeys.BROKER_CLIENT_ID) &&
+            clientId &&
+            correlationId
+        ) {
+            this.performanceClient?.addFields(
+                {
+                    embeddedClientId: clientId,
+                },
+                correlationId
+            );
+        }
+
         return queryParameterArray.join("&");
     }
+
+    addBrokerParameters(params: BrokerParameters): void {
+        const brokerParams: StringDict = {};
+        brokerParams[AADServerParamKeys.BROKER_CLIENT_ID] =
+            params.brokerClientId;
+        brokerParams[AADServerParamKeys.BROKER_REDIRECT_URI] =
+            params.brokerRedirectUri;
+
+        this.addClientId(params.embeddedClientId);
```

Comment:
Are we calling this function twice now?

Reply:
This was added as a safeguard. Removed this and added tests to make sure `clientId` is always set in the query string.